app/controllers/event_updates_controller.rb

```diff
@@ -1,31 +1,37 @@
 class EventUpdatesController < ApplicationController
   before_action :authenticate_user!, except: [:show]
   before_action :set_event_update, only: [:destroy]
+  before_action :set_event, only: :create
   before_action :access_allowed?
 
-
-
   def create
-    @event_update = EventUpdate.create(event_id: Event.friendly.find(params[:event_id]).id, details: params[:event_update][:details], user_id: current_user.id)
+    @event_update = @event.event_updates.create(event_updates_params)
   end
 
-
   def destroy
     # @eu_id = @event_update.id
     # @event_id = @event_update.event_id
     @event_update.destroy
   end
 
-
   private
 
-  def event_update_params
-    params.require(:event_update).permit(:details)
-  end
+    def event_update_params
```

Comment:
@GaganGupta19 
Rest of it looks cool, but then this method becomes redundant because of event_updates_params?
Can we rename that one to this and remove it?

Reply:
sure let me check it

Reply:
can we pass this user_id from view?

Reply:
Nope, we need it to be the current logged in user